src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -90,34 +90,32 @@
      * @param registryName The name of the registry, should include namespace. May come from another DeferredRegister through {@link #getRegistryName()}.
      * @param modid The namespace for all objects registered to this DeferredRegister
      */
-    public static <B extends IForgeRegistryEntry<B>> DeferredRegister<B> create(ResourceLocation registryName, String modid)
+    public static <B> DeferredRegister<B> create(ResourceLocation registryName, String modid)
     {
-        return new DeferredRegister<>(registryName, null, modid);
+        return new DeferredRegister<>(ResourceKey.createRegistryKey(registryName), null, modid);
     }
 
-    private final ResourceLocation registryName;
-    private final Class<T> superType;
+    private final ResourceKey<? extends Registry<T>> registryKey;
+    private final Class<? extends IForgeRegistryEntry<?>> superType;
```

Comment:
These should be marked as `@Nullable`. Additionally, add comments which denote which of the two are nullable at which time.